include/control_toolbox/rate_limiter.hpp

```diff
@@ -0,0 +1,306 @@
+   * If min_first_derivative_pos/max_first_derivative_neg values are NAN, symmetric limits are used
+   */
+  RateLimiter(
+    T min_value = NAN, T max_value = NAN,
+    T min_first_derivative_neg = NAN, T max_first_derivative_pos = NAN,
+    T min_first_derivative_pos = NAN, T max_first_derivative_neg = NAN,
+    T min_second_derivative = NAN, T max_second_derivative = NAN);
+
+  /**
+   * \brief Limit the value and first_derivative
+   * \param [in, out] v  value, e.g. [m/s]
+   * \param [in]      v0 Previous value to v , e.g. [m/s]
+   * \param [in]      v1 Previous value to v0, e.g. [m/s]
+   * \param [in]      dt Time step [s]
+   * \return Limiting factor (1.0 if none)
+   */
+  T limit(T & v, T v0, T v1, T dt);
+
+  /**
+   * \brief Limit the value
+   * \param [in, out] v value, e.g. [m/s]
+   * \return Limiting factor (1.0 if none)
+   */
+  T limit_value(T & v);
+
+  /**
+   * \brief Limit the first_derivative
+   * \param [in, out] v  value, e.g. [m/s]
+   * \param [in]      v0 Previous value, e.g. [m/s]
+   * \param [in]      dt Time step [s]
+   * \return Limiting factor (1.0 if none)
+   */
+  T limit_first_derivative(T & v, T v0, T dt);
+
+  /**
+   * \brief Limit the second_derivative
+   * \param [in, out] v  value, e.g. [m/s]
+   * \param [in]      v0 Previous value to v , e.g. [m/s]
+   * \param [in]      v1 Previous value to v0, e.g. [m/s]
+   * \param [in]      dt Time step [s]
+   * \return Limiting factor (1.0 if none)
+   * \see http://en.wikipedia.org/wiki/jerk_%28physics%29#Motion_control
+   */
+  T limit_second_derivative(T & v, T v0, T v1, T dt);
+
+
+  /**
+   * \brief Set the parameters
+   *
+   * \param [in] min_value Minimum value, e.g. [m/s], usually <= 0
+   * \param [in] max_value Maximum value, e.g. [m/s], usually >= 0
+   * \param [in] min_first_derivative_neg Minimum first_derivative, negative value, e.g. [m/s^2], usually <= 0
+   * \param [in] max_first_derivative_pos Maximum first_derivative, positive value, e.g. [m/s^2], usually >= 0
+   * \param [in] min_first_derivative_pos Asymmetric Minimum first_derivative, positive value, e.g. [m/s^2], usually <= 0
+   * \param [in] max_first_derivative_neg Asymmetric Maximum first_derivative, negative value, e.g. [m/s^2], usually >= 0
+   * \param [in] min_second_derivative Minimum second_derivative, e.g. [m/s^3], usually <= 0
+   * \param [in] max_second_derivative Maximum second_derivative, e.g. [m/s^3], usually >= 0
+   *
+   * \note
+   * If max_* values are NAN, the respective limit is deactivated
+   * If min_* values are NAN, defaults to -max if unspecified
+   * If min_first_derivative_pos/max_first_derivative_neg values are NAN, symmetric limits are used
+   */
+  void set_params(
+    T min_value = NAN, T max_value = NAN,
+    T min_first_derivative_neg = NAN, T max_first_derivative_pos = NAN,
+    T min_first_derivative_pos = NAN, T max_first_derivative_neg = NAN,
+    T min_second_derivative = NAN, T max_second_derivative = NAN);
```

Comment:
We should use the C++ `NaN` version and not C version.

Basically, the `NaN` should be changed to `std::numeric_limits<double>::quiet_NaN()` - and probably add `#include <limits>` file.

Reply:
done